### ISSUE TITLE
Infinite genome-wide lambda inflation and estimated genome-wide heritability 

### ISSUE THREAD

**Issue by marielohcs** [bvilhjal/ldpred#80]
Hi bvilhjal

I am trying to run ldpred twice, with the same reference and validation panel, but with different summary statistics based on different populations. 

For one set of summary statistics (N=40,000), ldpred ran fine with no errors.
However, for the other set of summary statistics (N=200,000), I was unable to run ldpred step (see output and error logs below).
I suspect it might be linked to the infinite genome-wide lambda inflation and estimated genome-wide heritability observed? 

Any thoughts on what might be causing this error? 
Many thanks in advance!

Cheers, 
Marie

Output (in output log): 
Loading LD information from file: (file location)
Applying LDpred with LD radius: 160
Genome-wide lambda inflation: inf Genome-wide mean LD score: 8.07900272708
Estimated genome-wide heritability: inf
Calculating LDpred-inf weights
Calculating scores for Chromosome 1
Calculating scores for Chromosome 2
Calculating scores for Chromosome 3
Calculating scores for Chromosome 4
Calculating scores for Chromosome 5
Calculating scores for Chromosome 6
Calculating scores for Chromosome 7
Calculating scores for Chromosome 8
Calculating scores for Chromosome 9
Calculating scores for Chromosome 10
Calculating scores for Chromosome 11
Calculating scores for Chromosome 12
Calculating scores for Chromosome 13
Calculating scores for Chromosome 14
Calculating scores for Chromosome 15
Calculating scores for Chromosome 16
Calculating scores for Chromosome 17
Calculating scores for Chromosome 18
Calculating scores for Chromosome 19
Calculating scores for Chromosome 20
Calculating scores for Chromosome 21
Calculating scores for Chromosome 22
Starting LDpred with p=1.0000

Error message (in error log):
LDpred.py:384: RuntimeWarning: invalid value encountered in double_scalars
  hdmp_hdmpn = (hdmp / hdmpn)
Traceback (most recent call last):
  File "LDpred.py", line 576, in <module>
    main()
  File "LDpred.py", line 571, in main
    ld_dict = ld_dict, n=p_dict['N'], num_iter=p_dict['num_iter'], h2=p_dict['H2'], verbose=False)
  File "LDpred.py", line 295, in ldpred_genomewide
    start_betas=LDpred_inf_chrom_dict[chrom_str], zero_jump_prob=zero_jump_prob)
  File "LDpred.py", line 395, in ldpred_gibbs
    rand_norms = stats.norm.rvs(0, (hdmp_hdmpn) * (1 / n), size=m)
  File "/apps/python/2.7.3/lib/python2.7/site-packages/scipy/stats/distributions.py", line 694, in rvs
    raise ValueError("Domain error in arguments.")
ValueError: Domain error in arguments.

**Comment by marielohcs** [comment]
P.S.: I've tried changing the sample size to 40,000 (same as my other dataset) but still had the same error. 
I've also attempted to manually set the heritability myself using the --H2 option (set to a heritability of around 0.2, which is what I got with my other dataset). This allowed the ldpred step to actually run, but I ended up with models with AUC=0.

**Comment by bvilhjal** [comment]
Thanks for this comment, and I apologize the slow reply.  

The lambda inflation factor seems rather extreme, possibly indicating a severe p-value inflation problem.  That could possibly explain the error you noted, i.e. the some term in the scaling becomes an invalid number due to computer float precision limits.  Have you looked at, e.g., a QQ plot for the GWAS p-values?

Best,
Bjarni

**Comment by marielohcs** [comment]
Thanks for your reply - sorry it took me a while to get back on this too! (actually I missed your reply somehow)
You were right - I had 16 SNPs with p-value=0 in my dataset. After removing these SNPs, the script now runs fine!

**Comment by marielohcs** [comment]
So, the script ran fine w/o errors after I removed the 16 SNPs with p-value=0. 
However, it appears that the model does not "converge" when I ran the validation. What I meant by this statement is that usually during validation, the AUC and R2 will increase as p (causal proportion) increases, maximises at a certain p value, and decreases as p further increases. However, I do not see that with this model. The AUC went up and down and then up again as p increases, which was v strange...
Any thoughts..?
